cpp/tests/geometry/AxisAlignedBoundingBox.cpp

```diff
@@ -0,0 +1,299 @@
+    box += box1;
+    ExpectEQ(box.min_bound_, v_t{0, 0, -1});
+    ExpectEQ(box.max_bound_, v_t{1, 1, 1});
+}
+
+TEST(AxisAlignedBoundingBox, PlusEquals4) {
+    AxisAlignedBoundingBox box({0, 0, 0}, {1, 1, 1});
+    AxisAlignedBoundingBox box1({0.25, 0.25, 0.25}, {2, 0.5, 0.5});
+    box += box1;
+    ExpectEQ(box.min_bound_, v_t{0, 0, 0});
+    ExpectEQ(box.max_bound_, v_t{2, 1, 1});
+}
+
+TEST(AxisAlignedBoundingBox, PlusEquals5) {
+    AxisAlignedBoundingBox box({0, 0, 0}, {1, 1, 1});
+    AxisAlignedBoundingBox box1({0.25, 0.25, 0.25}, {0.5, 2, 0.5});
+    box += box1;
+    ExpectEQ(box.min_bound_, v_t{0, 0, 0});
+    ExpectEQ(box.max_bound_, v_t{1, 2, 1});
+}
+
+TEST(AxisAlignedBoundingBox, PlusEquals6) {
+    AxisAlignedBoundingBox box({0, 0, 0}, {1, 1, 1});
+    AxisAlignedBoundingBox box1({0.25, 0.25, 0.25}, {0.5, 0.5, 2});
+    box += box1;
+    ExpectEQ(box.min_bound_, v_t{0, 0, 0});
+    ExpectEQ(box.max_bound_, v_t{1, 1, 2});
+}
+
+// IsEmpty permutations
+class IsEmptyTests : public TestWithParam<emp_t> {};
+
+TEST_P(IsEmptyTests, CheckBoxEmpty) {
+    AxisAlignedBoundingBox box(std::get<0>(GetParam()),
+                               std::get<1>(GetParam()));
+    EXPECT_EQ(std::get<2>(GetParam()), box.IsEmpty());
+}
+
+INSTANTIATE_TEST_CASE_P(AABBTests,
+                        IsEmptyTests,
+                        Values(emp_t{{4, 2, 3}, {4, 5, 6}, true},
+                               emp_t{{1, 5, 3}, {4, 5, 6}, true},
+                               emp_t{{1, 2, 6}, {4, 5, 6}, true},
+                               emp_t{{1, 2, 3}, {4, 5, 6}, false}));
+
+// Get Max Extent
+class MaxExtentTests : public TestWithParam<dbl_t> {};
+
+TEST_P(MaxExtentTests, CheckMaxExtent) {
+    AxisAlignedBoundingBox box(std::get<0>(GetParam()),
+                               std::get<1>(GetParam()));
+    EXPECT_DOUBLE_EQ(std::get<2>(GetParam()), box.GetMaxExtent());
+}
+
+INSTANTIATE_TEST_CASE_P(AABBTests,
+                        MaxExtentTests,
+                        Values(dbl_t{{-1, -2, -3}, {1, 2, 10}, 13},
+                               dbl_t{{-1, -2, -3}, {10, 2, 3}, 11},
+                               dbl_t{{-1, -2, -3}, {1, 10, 3}, 12}));
+
+// Volume
+class VolumeTests : public TestWithParam<dbl_t> {};
+
+TEST_P(VolumeTests, CheckMaxExtent) {
+    AxisAlignedBoundingBox box(std::get<0>(GetParam()),
+                               std::get<1>(GetParam()));
+    EXPECT_DOUBLE_EQ(std::get<2>(GetParam()), box.Volume());
+}
+
+INSTANTIATE_TEST_CASE_P(AABBTests,
+                        VolumeTests,
+                        Values(dbl_t{{-1, -2, -3}, {1, 2, 3}, 2 * 4 * 6},
+                               dbl_t{{0, 0, 0}, {1, 1, 1}, 1},
+                               dbl_t{{0, 0, 0}, {3, 1, 1}, 3},
+                               dbl_t{{0, 0, 0}, {1, 4, 1}, 4},
+                               dbl_t{{0, 0, 0}, {1, 1, 5}, 5}));
+
+}  // namespace tests
+}  // namespace open3d
```

Comment:
Please add an extra line at the end of the file.

Reply:
newline added